cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,1544 @@
+    state: absent
+    profile: boto3_profile_goes_here
+  register: vpc_peer
+
+# Complete example to reject a local account peering connection.
+# Boto3 profile for the other account must exist in ~/.aws/credentials
+- name: Create VPC peering Connection.
+  ec2_vpc_peer:
+    region: us-west-2
+    vpc_id: vpc-12345678
+    peer_vpc_id: vpc-87654321
+    state: present
+    resource_tags:
+      - Name: new_peer
+      - Env: development
+  register: vpc_peer
+
+- name: Reject a VPC peering Connection
+  ec2_vpc_peer:
+    region: us-west-2
+    peering_id: "{{ vpc_peer.vpc_peering_connection_id }}"
+    state: reject
+  register: vpc_peer
+
+# Complete example to reject a cross account peering connection.
+# Boto3 profile for the other account must exist in ~/.aws/credentials
+- name: Create cross account VPC peering Connection.
+  ec2_vpc_peer:
+    region: us-west-2
+    vpc_id: vpc-12345678
+    peer_vpc_id: vpc-87654321
+    state: present
+    peer_owner_id: 12345678910
+    resource_tags:
+      - Name: new_peer
+      - Env: development
+  register: vpc_peer
+
+- name: Reject a cross account VPC peering Connection
+  ec2_vpc_peer:
+    region: us-west-2
+    peering_id: "{{ vpc_peer.vpc_peering_connection_id }}"
+    state: reject
+    profile: boto3_profile_goes_here
+  register: vpc_peer
+'''
+RETURN = '''
+success:
+  description: Returns true if all succeeded and false if it failed.
+  returned: In all cases.
+  type: bool
+  sample: true
+changed:
+  description: Returns true if action made a changed  and false if it didn't.
+  returned: In all cases.
+  type: bool
+  sample: true
+status:
+  description: Dictionary containing the message and code.
+  returned: Success.
+  type: dictionary
+  sample:
+    {
+        "message": "Active",
+        "code": "active"
+    }
+tags:
+  description: List of dictionaries containing the key, val of each tag.
+  returned: Success.
+  type: list
+  sample:
+    [
+        {
+            "value": "web",
+            "key": "service"
+        }
+    ]
+accepter_vpc_info:
+  description: Dictionary containing the owner_id, vpc_id, and cidr_block.
+  returned: Success.
+  type: dictionary
+  sample:
+    {
+        "owner_id": "12345678910",
+        "vpc_id": "vpc-12345678",
+        "cidr_block": "172.31.0.0/16"
+    }
+vpc_peering_connection_id:
+  description: The peering connection id.
+  returned: Success.
+  type: string
+  sample: pcx-12345678
+requester_vpc_info:
+  description: Dictionary containing the owner_id, vpc_id, and cidr_block.
+  returned: Success.
+  type: dictionary
+  sample:
+    {
+        "owner_id": "12345678910",
+        "vpc_id": "vpc-12345678",
+        "cidr_block": "10.100.0.0/16"
+    }
+'''
+
+try:
+    import botocore
+    import boto3
+    import boto3.session
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import datetime
+import re
+
+def create_client_with_profile(profile_name, region, resource_name='ec2'):
```

Comment:
Is this necessary if using standard ansible connection method to aws? https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/GUIDELINES.md#boto3-1